File: sleep/ocaml_sleep/sleep.ml
(* comments *)
let seconds = 5;;


let main () =
  if  Array.length Sys.argv <> 1 then
    begin
      Printf.printf "%s <noargs>\n" Sys.argv.(0);
      exit 1
    end
  else
    begin
      print_endline "Sleeping...";
      Unix.sleep seconds;
      print_endline "Awake!"
    end

let _ = main ()

File: sleep/c_sleep/sleep.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <string.h>

int main( int argc, char *argv[] ) {
  unsigned int seconds = 10;

  if( argc != 1 ) {
    fprintf( stderr, "Usage: %s <noargs>\n", argv[0] );
    exit( 1 );
  }

  printf("Sleeping... for 10 seconds\n");
  sleep(seconds);
  printf("Awake!\n");
  return 0;
}
